packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -47,14 +47,43 @@ test.describe('Chat Composite E2E Tests', () => {
 
     await stubMessageTimestamps(page1);
     expect(await page1.screenshot()).toMatchSnapshot('received-messages.png');
-
     await waitForMessageSeen(page0);
+
     await stubMessageTimestamps(page0);
-    expect(await page0.screenshot()).toMatchSnapshot('read-message-status.png');
+    await page0.locator(dataUiId('chat-composite-message-status-icon')).click();
+    await page0.waitForSelector(dataUiId('chat-composite-message-tooltip'));
+    expect(await page0.screenshot()).toMatchSnapshot('read-message-tooltip-text.png');
+  });
+
+  test('page[0] can receive read receipt from page[1] and show it in contextual menu', async ({ pages }) => {
+    const testMessageText = 'How the turn tables';
+    const page0 = pages[0];
+    await sendMessage(page0, testMessageText);
+    await waitForMessageDelivered(page0);
+    await stubMessageTimestamps(page0);
+    expect(await page0.screenshot()).toMatchSnapshot('sent-messages.png');
+
+    const page1 = pages[1];
+    await waitForMessageWithContent(page1, testMessageText);
+
+    // It could be too slow to get typing indicator here, which makes the test flakey
```

Comment:
This is now in more than one place, can this be a util fn: `await waitForTypingIndicatorHidden(page)`?

Reply:
updated ! :)